### ISSUE TITLE
snapctl gives invalid character message without helpful information

### ISSUE THREAD

**Issue by mjbrender** [intelsdi-x/snap#717]
I have an instance of snap running locally and then run snapctl commands without success: 

```
build/bin/snapctl metric list
Error getting metrics: invalid character '<' looking for beginning of value
./snapctl task list
Error getting tasks:
invalid character '<' looking for beginning of value
./snapctl plugin list
Error: invalid character '<' looking for beginning of value
```

This was caused by running a proxy service that didn't let IPv6 traffic through. I think that means the proxy is passing HTML from a 404 page. 

A few thoughts after talking to @IRCody: 

* A better error would be helpful on invalid input 
* As a user, I would love to have some of the invalid input returned to me. **Some** being the operative word - bounding how much would be safe (and helpful from a logging / user POV)

**Comment by geauxvirtual** [comment]
What are you describing as invalid input? The command you ran was valid input. Having something passed to snapctl that wasn't from snapd is what is causing the error. snapctl should verify any response it receives came from snapd. If it didn't, it should throw an error saying it couldn't connect to snapd.

**Comment by mjbrender** [comment]
@geauxvirtual there's input somewhere, right? snapctl is reading the output from the proxy and erroring on the '<' character. +1 to wording it better though. Something more like "unable to reach snapd instance at [::1]:8181. Received ______" 

Note, this is a different error than you get when no snapd instance is running locally. With nothing running, I get: 

```
./build/bin/snapctl metric list
Error getting metrics: Get http://localhost:8181/v1/metrics/*?ver=0: read tcp [::1]:52038->[::1]:52037: read: connection reset by peer
```

I bring that up since it confirms this issue is more than whether a snapd instance is reached or not.

**Comment by IRCody** [comment]
I think having a better error message would be good. This comes from [here](https://github.com/intelsdi-x/snap/blob/master/mgmt/rest/client/client.go#L188-L189).

The marshaling fails with an error and that is passed along without context. We could do something like this as an error message:
```
> snapctl -u 'http://localhost:8080' plugin list
Error: Invalid API response:  invalid character 'N' looking for beginning of value.

 Received: <Part of the invalid response for debugging?>

exit status 1
```

**Comment by mjbrender** [comment]
:+1: awesome @IRCody. Thanks.

**Comment by IRCody** [comment]
Closing this since it was addressed in #718.
